lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1417,21 +1417,74 @@ func (w *WalletKit) fundPsbtInternalWallet(account string,
 				return err
 			}
 
+			eligible := make([]*lnwallet.Utxo, 0, len(utxos))
+			for _, utxo := range utxos {
+				// Don't use inputs which are unconfirmed and
+				// used by the sweeper.
+				if utxo.Confirmations == 0 &&
+					w.cfg.Sweeper.IsSweeperOutpoint(
+						utxo.OutPoint,
+					) {
+
+					log.Warnf("Cannot use unconfirmed "+
+						"utxo=%v because it is "+
+						"unstable and could be "+
+						"replaced", utxo.OutPoint)
+
+					continue
+				}
+				eligible = append(eligible, utxo)
+			}
+
 			// Validate all inputs against our known list of UTXOs
 			// now.
-			err = verifyInputsUnspent(packet.UnsignedTx.TxIn, utxos)
+			err = verifyInputsUnspent(
+				packet.UnsignedTx.TxIn, eligible,
+			)
 			if err != nil {
 				return err
 			}
 		}
 
+		// currentHeight is needed do determine whether the internal
+		// wallet utxo is still unconfirmed.
+		_, currentHeight, err := w.cfg.Chain.GetBestBlock()
+		if err != nil {
+			return fmt.Errorf("unable to retrieve current "+
+				"height: %v", err)
+		}
+
 		// We made sure the input from the user is as sane as possible.
 		// We can now ask the wallet to fund the TX. This will not yet
 		// lock any coins but might still change the wallet DB by
 		// generating a new change address.
 		changeIndex, err := w.cfg.Wallet.FundPsbt(
-			packet, minConfs, feeSatPerKW, account,
-			keyScope, strategy,
+			packet, minConfs, feeSatPerKW, account, keyScope,
+			strategy, func(utxo wtxmgr.Credit) bool {
+				// Wallet utxos which are unmined have a height
+				// of -1.
+				if utxo.Height != -1 &&
+					utxo.Height <= currentHeight {
+
+					// Confirmed utxos are always allowed.
+					return true
+				}
+
+				// Utxos used by the sweeper are not used for
+				// channel openings.
+				allowed := !w.cfg.Sweeper.IsSweeperOutpoint(
+					utxo.OutPoint,
+				)
+				if !allowed {
+					log.Warnf("Cannot use unconfirmed "+
+						"utxo=%v because it is "+
+						"unstable and could be "+
+						"replaced", utxo.OutPoint)
+				}
+
+				return allowed
+
+			},
```

Comment:
Can we not inline the closure definition at the argument site? I think it would be a good time to name the condition the closure is checking.